### PR TITLE
EZP-29601: As a editor I want to be able to insert html code in a richtext field

### PR DESCRIPTION
**Pull request by vidarl**
This one implements [EZP-29601](https://jira.ez.no/browse/EZP-29601)

### CODE REVIEW

File: lib/FieldType/XmlText/Converter/RichText.php [hunk 1]
```diff
@@ -503,13 +502,33 @@ protected function log($logLevel, $message, $context = [])
...
+    {
+        $xpath = new DOMXPath($document);
+
+        $xpathExpression = '//literal';
+
+        $elements = $xpath->query($xpathExpression);
+
+        foreach ($elements as $element) {
+            if ($element->getAttribute('class') !== 'html') {
+                $this->log(LogLevel::ERROR, "Only literal tag with class=\"html\" supported. Tag with different class removed where contentobject_attribute.id=$contentFieldId");
```
Comment (andrerom) [inline lib/FieldType/XmlText/Converter/RichText.php]: "supported" at the moment.

**Comment by andrerom** [comment]
Possible to get some more info in PR description on choices made here? For instance why did we need to escape utf8 characters?

**Comment by vidarl** [comment]
Sorry, my tests are the sources of the confusion

For PHPUnit, `&#xE1;` and `á` are equal when asserting XML. I have changed the tests to use the utf8 characters instead. It is indeed less confusing as the xml is written to db using the utf8 characters, not using the entities

characters like `<` and `>` are already converted to their corresponding xml entities (`&lt;` and `&gt;`) in ezxmltext, and are copied as-is into richtext document. No conversion needed in that point

File: lib/FieldType/XmlText/Converter/RichText.php (hunk 1 above)
Comment (vidarl) [reply lib/FieldType/XmlText/Converter/RichText.php]: fixed

**Pull request merged**